beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/path/aws/S3Client.java

```diff
@@ -50,11 +52,20 @@ void deleteObject(String bucket, String key) {
   }
 
   List<S3ObjectSummary> listObjects(String bucket, String key) {
```

Comment:
@patduin It does have to do with batch deletions here. `S3Client.listObjects(...)` now paginates through the results to return everything at a key.

Reply:
A possible issue with this implementation is that if we have a drop table and there are 10's or 100s of thousands of files, this could fail

Reply:
right I see I got distracted by the byte calculating stuff. Why would it fail?

Reply:
Possibly because the batching is within the `listObjects` method only, so this gets all of the objects first and processes them all second. Instead of getting a batch, processing them, and then getting the next batch and processing them. Potentially could be too heavy to handle it all at once.

Reply:
ok so you want to go with this approach first and if it fails we'll deal with it?

Reply:
Yep, this is the plan